onnxruntime/test/providers/cpu/tensor/gather_op_test.cc

```diff
@@ -76,7 +77,9 @@ TEST(GatherOpTest, Gather_invalid_index_cpu) {
   test.AddInput<int32_t>("indices", {3}, {0LL, 1L, 1000L});
   test.AddOutput<float>("output", {1}, {1.0f});
 
-  test.Run(OpTester::ExpectResult::kExpectFailure, "Mismatch between number of source and target dimensions.");
+  std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+  execution_providers.push_back(DefaultCpuExecutionProvider());
```

Comment:
Does this test only fail on CPU? 

If it's shape mismatch, it should fail on all EPs, right?

Reply:
There's separate test for GPU "TEST(GatherOpTest, Gather_invalid_index_gpu). In the test above, not all EPs failed at the same place, thus error message is inconsistent. The final outcome of failure/success and message from provider_test_utils therefore impossible to match. (usually the last one over-write previous EP outcome). Since the test name is _cpu, and there's already another test for gpu on other EPs, thus I updated it to run on CPU EP only.

Reply:
BTY, I can dup the test, to exclude CPU. so we have two tests covers all EPs. easy change.

Reply:
actually this test Gather_invalid_index_gpu is already doing that.